server/server_race_test.go

```
@@ -0,0 +1,181 @@
+				return next(ctx, req)
+			}
+		}
+		WithToolHandlerMiddleware(middleware)(srv)
+	})
+
+	runConcurrentOperation(&wg, testDuration, "list-tools", func() {
+		srv.handleListTools(ctx, "123", mcp.ListToolsRequest{})
+	})
+
+	runConcurrentOperation(&wg, testDuration, "list-prompts", func() {
+		srv.handleListPrompts(ctx, "123", mcp.ListPromptsRequest{})
```

golangci-lint (1.64.8)

Error return value of `srv.handleListPrompts` is not checked

(errcheck)